gbasis/wrappers.py

```diff
@@ -203,6 +199,22 @@ def angmom_components_sph(self):
 
             coeffs = exps_coeffs[:, 1:]
 
-            basis.append(PyscfShell(angmom, np.array(coord), np.array(coeffs), np.array(exps)))
+            if type(coord_types) == str:
+            # if coord_types given as a single string, assign the specified type to all contractions for all atoms
+                if coord_types == "spherical":
+                    basis.append(PyscfShell(angmom, np.array(coord), np.array(coeffs), np.array(exps), 'p'))
+                elif coord_types == "cartesian":
+                    basis.append(PyscfShell(angmom, np.array(coord), np.array(coeffs), np.array(exps), 'c'))
+                else:
+                    raise ValueError("If coord_types is a string, it must be either 'spherical' or 'cartesian'.")
+            elif type(coord_types) == list:
+                # if coord_types given as a list, assign the specified type to each atom's contractions individually
+                if len_coord_types == mol.nbas:
+                    basis.append(PyscfShell(angmom, np.array(coord), np.array(coeffs), np.array(exps), coord_types.pop(0)))
```

Comment:
@maximilianvz, the problem with popping elements from `coord_types` list is that, if there is more than 1 atom in the molecule, the `coord_types` list will be empty after the shells for the first atom are appended to `basis`. This tentative bug was not caught by the existing tests, because they all had a string `coord_type`.

Considering the recent change in 729d83e92d19e87b545e5aff64b575639ea449b2, and the fact that PySCF does not support mixed coordinate types (based on my quick look), testing this can be tricky (any thoughts?). Then, we should either only support `str` type `coord_types` or index the `coord_types` list (instead of popping). The latter is less desitable, because I am not still sure how to test a list `coord_types`.

@leila-pujal, @RichRick1, @msricher, and @PaulWAyers, do you know whether PySCF supports mixed "cartesian" and "spherical" basis?

Reply:
From what I remember using PySCF, it does not allow for mixed cartesian/spherical. There's an attribute in the `Mole` class called `cart` that's boolean. `True` the basis is cartesian and `False` spherical.

Reply:
@FarnazH, if PySCF doesn't allow for mixed cartesian/spherical coordinates, then we should remove the `elif type(coord_types) == list:` clause altogether, as the only time you would want to pass a list here is if you were using a mix of these coordinate systems. By adding [this line](https://github.com/maximilianvz/gbasis/blob/610dd8f8d9777244ba7a7836293cf9762793a30e/gbasis/wrappers.py#L173), you prevent `coord_types` from being a list anyway. Thus, I don't see any "bug" applying to `from_pyscf` if we were to remove that part of the code. 

However, [`make_contractions`](https://github.com/maximilianvz/gbasis/blob/610dd8f8d9777244ba7a7836293cf9762793a30e/gbasis/parsers.py#L168) does allow for mixed coordinate systems. It also uses this `elif type(coord_types) == list:` clause, but I believe [this test](https://github.com/maximilianvz/gbasis/blob/610dd8f8d9777244ba7a7836293cf9762793a30e/tests/test_overlap_asymm.py#L58C8-L58C8) shows that popping from the `coord_types` list will work. In that example of a Krypton dimer, each Krypton atom contributes 5 shells, for a total of 10 shells. In the call to `make_contractions`, a list of length 10, where all but the last entries are "spherical", is passed. [You will only ever `pop` from `coord_types` if the initial length of `coord_types` is equal to the TOTAL number of contractions in the molecule](https://github.com/maximilianvz/gbasis/blob/610dd8f8d9777244ba7a7836293cf9762793a30e/gbasis/parsers.py#L225C17-L225C17). In this case, the onus is on the user to specify a list of length 10 to `make_contractions`, and to ensure that they know that the first 5 coordinate types in this list would get applied to the 5 shells of the first Krypton atom, while the last 5 coordinate types in that list (where the last element was specified as "cartesian") would get applied to the five shells of the second Krypton atom.  

Please let me know if you have any questions or doubts.

Reply:
@maximilianvz, based on our discussions, I made the changes below. I believe this is good to be merged, let me know if you have any comments.

Take a look at the commits on how the code is refactored to avoid nested `if/elif/else` statements. Also, when checking the type of a variable, it is more appropriate to use `isinstance` function.